### DISCUSSION TITLE
Foundry[Forge Init] - Missing "lib" Folder in Foundry Directory .

### DISCUSSION THREAD

**Discussion by vigneshbalu96** [Cyfrin/foundry-full-course-cu#1134] [Q&A]
Hi Blockchaineers, :hand: 

I recently started working with Foundry and encountered an issue while executing "forge init." Here's a brief history of the problem:

:thought_balloon: Initially, I mistakenly ran "forge init" in my home directory but later realised my error and tried to execute it in a separate Foundry directory (excluding the "lib" folder[not created by default]). Although I faced some issues initially, I managed to troubleshoot them. However, instead of deleting the folder structure created by the initial "forge init" command in my home directory, I mistakenly deleted the structure in the Foundry directory. Consequently, I cleared all files and folders in both the home and Foundry directories. :fire: 

![Screenshot from 2023-11-22 11-02-00](https://github.com/Cyfrin/foundry-full-course-f23/assets/107097800/741575c5-646a-43b8-a718-d712107ff121)

:point_right: Now, when I run "forge init" in the Foundry directory, I encounter an error. Although it creates the folder structure as intended, the "lib" folder is missing from the project's start. Have I overlooked anything in this process? :thinking: 

Thanks in advance for your assistance! :point_down:

**Comment by vigneshbalu96** [comment]
Used this command to create a lib folder but the folder seems to be empty.
```
forge install foundry-rs/forge-std
```
IDK why(Think it won't push this to the GitHub), when I did 
```
forge init --no-commit --force
```
It worked and started cloning into the lib, populating the sub-modules and registered for the lib path. 
Would like to know the other ways to populate the lib simultaneously so that I can also push it to the GitHub.

Any help would be appreciated, thanks!

**Reply by xKeywordx** [reply]
Give this guy a medal, this solved it for me 

forge init --no-commit --force

**Reply by SNERUOL** [reply]
Yes I also needed this one. Thanks!

**Reply by codeplay22** [reply]
but it also creates  "counter.sol "   file in   " src and script folder ".
Is That okay?

**Comment by Niferu** [comment]
Hello @vigneshbalu96 

Whenever you run `forge init` it will create bunch of folders as your starting template for your project. If there is any folder already created in desired directory this command will not work. As you are starting fresh project just do below:

Before:: Run command: `foundryup` -> this will update your foundry tool

1. Create new folder for your project or delete everything from your current project folder. (To create new project with foundry you just need clean empty folder)
2. Once you have clean folder just navigate to it in bash and run command `forge init .` -> this dot at the end will create all necessary filed/folders in that clean folder you are already in.

In this case you should not be missing any lib or any other folders/files if you will still tho just let me know.

In case you have removed some of your work you should be able to restore it from bin and if not you have to start over again.
